src/renderer/src/gl_scene_viewport.cpp

```diff
@@ -0,0 +1,162 @@
+	camera_controller_ = std::make_unique<SphericalCameraController>(
+			std::move(camera), 100.0f, glm::pi<float>());
+	heatmap_material_ = std::make_unique<HeatmapMaterial>();
+}
+
+void GLSceneViewport::DebugInit(const int /*w*/, const int /*h*/) {
+	data_ = std::make_unique<Impl>();
+	data_->axes->ApplyScale(4);
+	const Core::Shapes shapes = {
+			std::make_shared<BasicShape>(
+					ShapeId(0, 0), 0, Box3D({{0.0f, 1.0f}, {0.0f, 2.0f}, {0.0f, 3.0f}}
+             ))
+  };
+	scene_->AddShapes(shapes);
+	constexpr glm::vec3 offset = {0.1, 0.1, 0.1};
+	scene_->shapes()[0]->Translate(offset);
+}
+
+void GLSceneViewport::ClearResources() { ClearResourcesImpl(); }
+
+void GLSceneViewport::ClearResourcesImpl() {
+	heatmap_material_.reset();
+	scene_->Clear();
+	LOG_DEBUG("Context cleared");
+}
+
+void GLSceneViewport::RenderFrame() {
+	auto api = RendererAPI::instance();
+	api->Clear();
+
+	const ICamera& camera = camera_controller_->camera();
+
+	for (const auto& shape : scene_->shapes()) {
+		heatmap_material_->Use(shape->transform(), camera.viewProjectionMatrix());
+		api->DrawIndexed(shape->vertex_array());
+	}
+
+	if (data_) {
+		const auto& axes = data_->axes;
+		data_->debug_material->Use(axes->transform(), camera.viewProjectionMatrix(),
+		                           1.0f);
+		api->DrawIndexed(axes->vertex_array(), PrimitiveType::LINES);
+	}
+}
+
+void GLSceneViewport::Resize(const int w, const int h) {
+	RendererAPI::instance()->SetViewPort(0, 0, w, h);
+	camera_controller_->SetCameraAspectRatio(static_cast<float>(w) /
+	                                         static_cast<float>(h));
+}
+
+void GLSceneViewport::SetTemperatureRange(const float min, const float max) {
+	heatmap_material_->SetTemperatureRange(min, max);
+}
+
+void GLSceneViewport::SetColorRange(const ISceneViewport::Color min,
+                                    const ISceneViewport::Color max) {
+	heatmap_material_->SetColorRange({min[0], min[1], min[2]},
+	                                 {max[0], max[1], max[2]});
+}
+
+void GLSceneViewport::MoveCamera(const Vec2D /*screenPoint*/,
+                                 const Vec2D delta) {
+	static constexpr float zoom_coefficient = 0.002f;
+	const float zoom_level = camera_controller_->camera().zoom_level();
+
+	const float x_coefficient = -zoom_level * zoom_coefficient;
+	const float y_coefficient = zoom_level * zoom_coefficient;
+
+	const auto dx = static_cast<float>(delta.x);
+	const auto dy = static_cast<float>(delta.y);
+
+	const auto horizontal_translation =
+			x_coefficient * dx * camera_controller_->camera().rightVector();
+	const auto vertical_translation =
+			y_coefficient * dy * camera_controller_->camera().upVector();
+
+	camera_controller_->Translate(horizontal_translation + vertical_translation);
+}
+
+void GLSceneViewport::RotateCamera(const Vec2D /*screenPoint*/,
+                                   const Vec2D delta) {
+	static constexpr float zoom_coefficient = 1.0f / 1500.0f;
```

Comment:
as I remember you have special constants for it
`constexpr float zoom           = 15.0f;
constexpr float near_plane     = 1.0f;`
Please use them